languages/ocaml.scm

```diff
@@ -1100,6 +1104,21 @@
   )
 )
 
+(attribute_payload
+  (function_type
+    "->" @append_indent_start
+    (_) @append_indent_end
+    .
+  )
+)
+
+; Make an indented block where a function/match starts in PPX syntax.
+; NOTE This is probably a bit of a hack...
+(expression_item
+  .
+  (_) @prepend_indent_start
+) @append_indent_end
+
```

Comment:
I would say we can solve all three cases (functions, matches, constructed types) with something like
```scheme
(extension
  "[%" @append_indent_start
  "]" @prepend_indent_end
)
```
Does it work?

Reply:
We could even add `@prepend_empty_softline` on the closing bracket for good measure.

Reply:
That does work :+1: ...and I think with the softline before the closing bracket, the output looks nicer (imo):

```ocaml
let x =
  [%expr function
    | false -> 0.
    | true -> 1.
  ]
```

There's some double indentation though:
```ocaml
let h =
  [%madcast: float ->
      bool
  ]
```

See 5ae369c37bfe